docs/20_smart-contracts/10_getting-started/50_read-data-from-tables.md

```diff
@@ -0,0 +1,264 @@
+---
+title: Get Data from EOS Blockchain
+---
+
+## Prerequisites
+
+To follow this guide, you will need:
+
+- An understanding of the EOS blockchain and how it works.
+- A command-line interface to run curl commands.
+- Access to an EOS node or an EOS API service.
+
+## EOS Tables
+
+EOS stores data in tables, which are similar to database tables. Each table has a name and a set of fields. Tables are organized into scopes, which are defined by the smart contract that created the table.
+
+To retrieve data from a table, you need to know its name, scope, and the name of the smart contract that created it. You can also specify a lower and upper bound to limit the amount of data returned.
+
+## Methods to Retrieve Data from EOS Tables
+
+### Use get_table_rows Function
+
+The `get_table_rows` function retrieves rows from a table. It takes the following parameters in JSON format:
+
+- `"code"`: the eos account name which is the owner of the smart contract that created the table.
+- `"scope"`: the scope of the table, it is an eos account name.
+- `"table"`: a string representing the name of the table.
+- `"json"`: (optional) a boolean value that specifies whether to return the row results in JSON format or binary format, defaults to binary.
+- `"lower_bound"`: (optional) a string representing the lower bound for the table key, defaults to first value of the index used.
+- `"upper_bound"`: (optional) a string representing the upper bound for the table key, defaults to the last value of the index used.
+- `"index_position"`: (optional) the position of the index to use if the table has multiple indexes, accepted values are `primary`, `secondary`, `tertiary`, `fourth`, `fifth`, `sixth`, `seventh`, `eighth`, `ninth` , `tenth`, defaults to `primary`.
+- `"key_type"`: (optional) a string representing the type of the table key, supported values `i64`, `i128`, `i256`, `float64`, `float128`, `sha256`, `ripemd160`, `name`.
+- `"encode_type"`: (optional) a string representing the encoded type of the key_type parameter, either `dec` or `hex`, defaults to `dec`.
+- `"limit"`: limits the number of results returned, defaults to 10.
+- `"time_limit_ms"`: (optional) the maximum time should spend to retrieve the results, defaults to 10ms.
+- `"reverse"`: (options) if `true` the results are retrieved in reverse order, from lower_bound up towards upper_bound, defaults to `false`.
+
+The below example retrieves from the `producers` table, owned by the `eosio` account and having as `scope` the `eosio` name, the first three rows that have the primary key value (which is the producers account name) higher than or equal with the `argentinaeos`:
```

Comment:
We should use a different target for this. 

The `producers` table is a moving target. Would be better to use something like fetching the balance of `eosio` from the `eosio.token` contract instead as we can ensure that the result we are looking for will always exist